boards: nrf52840 common flash partition dtsi

jacobw opened the pull request:
Add common default flash partition layout for nrf52840 as many boards have used identical flash layouts.

The default flash layout was updated to remove scrach in 2022 (9a84258) but almost all boards were still using the previous layout, so this updates them to the new layout with allows for larger applications.

jacobw commented:
tests failed with these errors:

```
Error: Error: pod failed to come online with error: Error: Pod zrv2-linux-x64-4xlarge-cnx-bwqtb-runner-b98nd-workflow is unhealthy with phase status Pending
Error: Process completed with exit code 1.
Error: Executing the custom container implementation failed. Please contact your self hosted runner administrator.
<snip>
Error: Error: failed to run script step: TypeError: Cannot read properties of null (reading 'jobPod')
Error: Process completed with exit code 1.
Error: Executing the custom container implementation failed. Please contact your self hosted runner administrator.
```

Is this a CI failure? If so how can i resolve?

rerickson1 reviewed (dismissed):
Ezurio board looks good.

jacobw commented:
Amended to incorporate changes suggested by @nordicjm in [77791](https://github.com/zephyrproject-rtos/zephyr/pull/77791#discussion_r1740557411)

anangl commented on dts/common/nordic/nrf52840_flash.dtsi:
```diff
@@ -0,0 +1,48 @@
+};
+
+&flash0 {
+
```
nit: unnecessary blank line

anangl commented on dts/common/nordic/nrf52840_flash.dtsi:
Perhaps it would be better to name this file `nrf52840_partition.dtsi` for consistency with the stuff introduced by #74707.

jacobw replied:
I was wondering about that, and didn't as the other PR was looking at SRAM. But I agree and will change it.

I'll also therefore add sram to chosen too.

anangl reviewed (dismissed):
Looks good. Just two nitpicking comments.

nordicjm commented on dts/common/nordic/nrf52840_partition.dtsi:
```diff
@@ -4,8 +4,9 @@
  */
 
-/ {
+ / {
```
indent should go back to how it was

jacobw replied:
sorry I'm undoing a broken merge conflict...

jacobw replied:
fixed

nordicjm commented on dts/common/nordic/nrf52840_partition.dtsi:
```diff
@@ -4,8 +4,9 @@
-/ {
+ / {
 	chosen {
+        zephyr,sram = &sram0;
```
wrong indent

jacobw replied:
fixed

nordicjm commented on dts/common/nordic/nrf52840_partition.dtsi:
```diff
@@ -16,7 +17,6 @@
 		compatible = "fixed-partitions";
 		#address-cells = <1>;
 		#size-cells = <1>;
-
```
keep newline gap here

jacobw replied:
fixed

nordicjm commented on dts/common/nordic/nrf52840_partition.dtsi:
```diff
@@ -34,7 +34,6 @@
 		 * The flash starting at 0x000f8000 and ending at
 		 * 0x000fffff is reserved for use by the application.
 		 */
-
```
and here

jacobw replied:
fixed

jacobw commented:
Rebased after conflict with merged PR #77791.

jacobw committed:
boards: nrf52840 common partition dtsi

Add common default flash partition layout for nrf52840 as many boards
have used identical flash layouts.

The default flash layout was updated to remove scrach in 2022 (9a84258)
but almost all boards were still using the previous layout, so this
updates them to the new layout with allows for larger applications.

This commit also incorporates feeedback from @nordicjm in PR #77791 to
change slot0 to 0x00077000 and slot1 to 0x00075000: "If you use swap
using move, you need a sector for the data to be moved up by, and you
need space for the swap status fields, which is about a sector, so by
making the changes here you get the full 0x65000 for an image, without
these changes you get 0x64000.

Signed-off-by: Jacob Winther <jacob@9.nz>

jacobw commented:
Ping @rerickson1
